src/AWS.jl

```diff
@@ -360,6 +361,28 @@ function _return_headers(args::AbstractDict{String, <:Any})
     return return_headers
 end
 
+function _flatten_query(service::String, query::AbstractDict{String, <:Any}, result::Vector{Pair{String, String}}=Vector{Pair{String, String}}(), prefix::String="")
```

Comment:
This is good, but the Julia style for mutating/non-mutating function versions would look like this:

```suggestion
function _flatten_query(service::String, query::AbstractDict{String, <:Any}, prefix::String="")
    return _flatten_query!(Pair{String, String}[], service, query, prefix)
end

function _flatten_query!(result::Vector{Pair{String, String}}, service::String, query::AbstractDict{String, <:Any}, prefix::String)
```